lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs

```diff
@@ -2268,6 +2310,200 @@ prop_balanceTransactionUnresolvedInputs
         let dropped = map snd $ filter (not . fst) $ zip shouldKeep inputs
         pure (PartialTx tx inputs' redeemers, dropped)
 
+data BalanceTxGolden =
+    BalanceTxGoldenSuccess
+        { _walletCoin :: Coin
+        , _balancedFee :: Cardano.Lovelace
+        , _balancedMinFee :: Cardano.Lovelace
+        }
+    | BalanceTxGoldenFailure Coin String
+
+-- CSV with the columns: wallet_balance,fee,minfee,error
+instance Buildable BalanceTxGolden where
+    build (BalanceTxGoldenFailure c err) = mconcat
+        [ build c
+        , ",,,"
```

Comment:
Question: why three commas here? 😄

Reply:
I know, it looks strange! The idea was to have well-defined CSV columns for the sake of `gnuplot`. However, leaving empty values didn't seem to work... Maybe I should use a single comma unless we come up with a proper checked-in gnuplot script.

Reply:
-> ece4adc87, not sure what you think.

On a related note: having a concise `ToText` for balanceTx/coin-selection errors would be very neat!

Reply:
>On a related note: having a concise ToText for balanceTx/coin-selection errors would be very neat!

Perhaps rendering these as JSON makes more sense? Since these errors sometimes have to contain structured information.